python/TestHarness/schedulers/RunPBS.py

```diff
@@ -72,8 +76,157 @@ def updateParams(self, tester):
         minutes = int(int(tester.specs['max_time']) / 60) % 60
         self.specs['walltime'] = '{0:02d}'.format(hours) + ':' + '{0:02d}'.format(minutes) + ':00'
 
+        # Add PBS project directive
         if self.options.project:
             self.specs['pbs_project'] = '#PBS -P %s' % (self.options.project)
 
+        # Add PBS queue directive
         if self.options.queue:
             self.specs['pbs_queue'] = '#PBS -q %s' % (self.options.queue)
+
+    # Handle statuses supplied by qsub output
+    def handleQsubOutput(self, tester, output):
+        queue_job_id = 0
+        if 'command not found' in output:
+            tester.setStatus('QSUB NOT FOUND', tester.bucket_fail)
+        else:
+            test_output = os.path.join(self.specs['working_dir'], self.specs['job_name'])
+            job_id = re.search(r'^(\d+)\.[\W\w]+$', output)
+
+            # Job was launched and we have a PBS Job ID
+            if job_id:
+                queue_job_id = job_id.group(1)
+                tester.setStatus('%s LAUNCHED' % (str(queue_job_id)), tester.bucket_pending)
+
+            # Something went wrong trying to launch the job
+            else:
+                tester.setStatus('QSUB INVALID RESULTS: %s' % (output), tester.bucket_fail)
+
+        # Update the queue_data global so that it can be saved/retreived to/from the json file.
+        self.queue_data[tester.specs['test_name']] = { 'test_name'     : tester.specs['test_name'],
+                                                       'input_name'    : self.options.input_file_name,
+                                                       'std_out'       : test_output + '.o%s' % (queue_job_id),
+                                                       'std_err'       : test_output + '.e%s' % (queue_job_id),
+                                                       'test_dir'      : self.specs['working_dir'],
+                                                       'status_text'   : tester.getStatusMessage(),
+                                                       'id'            : queue_job_id }
+        # Write the json data to queue file
+        json.dump(self.queue_data, self.queue_file, indent=2)
+
+        return output
+
```

Comment:
This whole method should be in the base class... In fact I think all that belongs here is an re.complie of how to discover the returned job ID...